src/modules/systemlib/battery.cpp

```diff
@@ -239,12 +230,8 @@ Battery::computeScale()
 	// reusing capacity calculation to get single cell voltage before drop
 	const float bat_v = _param_v_empty.get() + (voltage_range * _remaining_voltage);
 
-	_scale = _param_v_full.get() / bat_v;
-
-	if (_scale > 1.3f) { // Allow at most 30% compensation
-		_scale = 1.3f;
+	// Allow at most 30% compensation
+	// Shouldn't ever be more than the power at full battery
+	_scale = math::constrain(_param_v_full.get() / bat_v, 1.0f, 1.3f);
```

Comment:
Thanks, it definitely doesn't. Makes me wonder if we might as well constrain NANs as well? The controllers are filled constrain, but finite checks are done inconsistently.

Fixed and rebased.